dnf/base.py

```diff
@@ -656,12 +643,8 @@ def resolve(self, allow_erasing=False):
                 exc = dnf.exceptions.Error(msg)
 
         if exc is not None:
-            if self._group_persistor:
-                self._group_persistor._rollback()
```

Comment:
An exception is being raised right after the rollback - history object will be destroyed along with GroupPersistor which holds the temporary changes.